### ISSUE TITLE
`cachix` not found, breaking install step of action

### ISSUE THREAD

**Issue by andschwa** [purcell/setup-emacs#33]
Hey @purcell, sadly an issue seems to have cropped up today. Not sure if it's a Nix update or a GitHub Action image runner update, but the install step fails because `cachix` isn't found:

```
2023-02-28T22:12:34.7268057Z ##[debug]Evaluating condition for step: 'Install Emacs'
2023-02-28T22:12:34.7269795Z ##[debug]Evaluating: success()
2023-02-28T22:12:34.7270126Z ##[debug]Evaluating success:
2023-02-28T22:12:34.7270521Z ##[debug]=> true
2023-02-28T22:12:34.7270876Z ##[debug]Result: true
2023-02-28T22:12:34.7271533Z ##[debug]Starting: Install Emacs
2023-02-28T22:12:34.7287606Z ##[debug]Loading inputs
2023-02-28T22:12:34.7288463Z ##[debug]Loading env
2023-02-28T22:12:34.7292773Z ##[group]Run purcell/setup-emacs@v3.0
2023-02-28T22:12:34.7292999Z with:
2023-02-28T22:12:34.7293187Z   version: 28.1
2023-02-28T22:12:34.7293371Z env:
2023-02-28T22:12:34.7293578Z   DOTNET_ROOT: /usr/share/dotnet
2023-02-28T22:12:34.7293801Z ##[endgroup]
2023-02-28T22:12:34.9132962Z   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2023-02-28T22:12:34.9138831Z                                  Dload  Upload   Total   Spent    Left  Speed
2023-02-28T22:12:34.9139380Z 
2023-02-28T22:12:35.0725764Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2023-02-28T22:12:35.0726569Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2023-02-28T22:12:35.1475979Z 
2023-02-28T22:12:35.1492646Z 100  4052  100  4052    0     0  17261      0 --:--:-- --:--:-- --:--:-- 17261
2023-02-28T22:12:35.1563226Z downloading Nix 2.14.0 binary tarball for x86_64-linux from 'https://releases.nixos.org/nix/nix-2.14.0/nix-2.14.0-x86_64-linux.tar.xz' to '/tmp/nix-binary-tarball-unpack.f9xXVn5ovk'...
2023-02-28T22:12:35.1618170Z   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2023-02-28T22:12:35.1619621Z                                  Dload  Upload   Total   Spent    Left  Speed
2023-02-28T22:12:35.1620044Z 
2023-02-28T22:12:35.3243690Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2023-02-28T22:12:35.3622425Z  50 20.2M   50 10.2M    0     0  62.7M      0 --:--:-- --:--:-- --:--:-- 62.6M
2023-02-28T22:12:35.3623204Z 100 20.2M  100 20.2M    0     0   100M      0 --:--:-- --:--:-- --:--:--  100M
2023-02-28T22:12:37.3053968Z Note: a multi-user installation is possible. See https://nixos.org/manual/nix/stable/installation/installing-binary.html#multi-user-installation
2023-02-28T22:12:37.3063362Z Warning: the flag --darwin-use-unencrypted-nix-store-volume
2023-02-28T22:12:37.3063761Z          is no longer needed and will be removed in the future.
2023-02-28T22:12:37.3063933Z 
2023-02-28T22:12:37.3074434Z [1;31mSwitching to the Multi-user Installer[0m
2023-02-28T22:12:37.3178173Z [32mWelcome to the Multi-User Nix Installation
2023-02-28T22:12:37.3188064Z [0m
2023-02-28T22:12:37.3188517Z This installation tool will set up your computer with the Nix package
2023-02-28T22:12:37.3188943Z manager. This will happen in a few stages:
2023-02-28T22:12:37.3189181Z 
2023-02-28T22:12:37.3189531Z 1. Make sure your computer doesn't already have Nix. If it does, I
2023-02-28T22:12:37.3189980Z    will show you instructions on how to clean up your old install.
2023-02-28T22:12:37.3190231Z 
2023-02-28T22:12:37.3190488Z 2. Show you what I am going to install and where. Then I will ask
2023-02-28T22:12:37.3190846Z    if you are ready to continue.
2023-02-28T22:12:37.3191062Z 
2023-02-28T22:12:37.3191333Z 3. Create the system users and groups that the Nix daemon uses to run
2023-02-28T22:12:37.3191668Z    builds.
2023-02-28T22:12:37.3191855Z 
2023-02-28T22:12:37.3192106Z 4. Perform the basic installation of the Nix files daemon.
2023-02-28T22:12:37.3192356Z 
2023-02-28T22:12:37.3192626Z 5. Configure your shell to import special Nix Profile files, so you
2023-02-28T22:12:37.3192972Z    can use Nix.
2023-02-28T22:12:37.3193153Z 
2023-02-28T22:12:37.3193331Z 6. Start the Nix daemon.
2023-02-28T22:12:37.3193539Z 
2023-02-28T22:12:37.3194024Z [32m[4;32mWould you like to see a more detailed list of what I will do?
2023-02-28T22:12:37.3198004Z [0mNo TTY, assuming you would say yes :)
2023-02-28T22:12:37.3199336Z 
2023-02-28T22:12:37.3199762Z I will:
2023-02-28T22:12:37.3199979Z 
2023-02-28T22:12:37.3200291Z  - make sure your computer doesn't already have Nix files
2023-02-28T22:12:37.3200688Z    (if it does, I will tell you how to clean them up.)
2023-02-28T22:12:37.3201160Z  - create local users (see the list above for the users I'll make)
2023-02-28T22:12:37.3201567Z  - create a local group (nixbld)
2023-02-28T22:12:37.3201917Z  - install Nix in to /nix
2023-02-28T22:12:37.3202302Z  - create a configuration file in /etc/nix
2023-02-28T22:12:37.3202778Z  - set up the "default profile" by creating some Nix-related files in
2023-02-28T22:12:37.3203126Z    /root
2023-02-28T22:12:37.3219770Z  - back up /etc/bash.bashrc to /etc/bash.bashrc.backup-before-nix
2023-02-28T22:12:37.3220310Z  - update /etc/bash.bashrc to include some Nix configuration
2023-02-28T22:12:37.3234068Z  - load and start a service (at /etc/systemd/system/nix-daemon.service
2023-02-28T22:12:37.3234629Z    and /etc/systemd/system/nix-daemon.socket) for nix-daemon
2023-02-28T22:12:37.3234901Z 
2023-02-28T22:12:37.3239035Z [32m[4;32mReady to continue?
2023-02-28T22:12:37.3239491Z [0mNo TTY, assuming you would say yes :)
2023-02-28T22:12:37.3268913Z 
2023-02-28T22:12:37.3269624Z [34m---- let's talk about sudo -----------------------------------------------------
2023-02-28T22:12:37.3279310Z [0mThis script is going to call sudo a lot. Normally, it would show you
2023-02-28T22:12:37.3279821Z exactly what commands it is running and why. However, the script is
2023-02-28T22:12:37.3280228Z run in a headless fashion, like this:
2023-02-28T22:12:37.3280457Z 
2023-02-28T22:12:37.3280734Z   $ curl -L https://nixos.org/nix/install | sh
2023-02-28T22:12:37.3280967Z 
2023-02-28T22:12:37.3281309Z or maybe in a CI pipeline. Because of that, I'm going to skip the
2023-02-28T22:12:37.3281721Z verbose output in the interest of brevity.
2023-02-28T22:12:37.3281955Z 
2023-02-28T22:12:37.3282124Z If you would like to
2023-02-28T22:12:37.3282658Z see the output, try like this:
2023-02-28T22:12:37.3282881Z 
2023-02-28T22:12:37.3283179Z   $ curl -L -o install-nix https://nixos.org/nix/install
2023-02-28T22:12:37.3283566Z   $ sh ./install-nix
2023-02-28T22:12:37.3283773Z 
2023-02-28T22:12:37.3283869Z 
2023-02-28T22:12:37.3284188Z [32m~~> Checking for artifacts of previous installs
2023-02-28T22:12:37.3290735Z [0mBefore I try to install, I'll check for signs Nix already is or has
2023-02-28T22:12:37.3291115Z been installed on this system.
2023-02-28T22:12:37.3329369Z 
2023-02-28T22:12:37.3329954Z [34m---- Nix config report ---------------------------------------------------------
2023-02-28T22:12:37.3330414Z [0m[1m        Temp Dir[0m:	/tmp/tmp.za2LMpim6x
2023-02-28T22:12:37.3330728Z [1m        Nix Root[0m:	/nix
2023-02-28T22:12:37.3331116Z [1m     Build Users[0m:	4
2023-02-28T22:12:37.3331486Z [1m  Build Group ID[0m:	30000
2023-02-28T22:12:37.3331867Z [1mBuild Group Name[0m:	nixbld
2023-02-28T22:12:37.3332097Z 
2023-02-28T22:12:37.3332324Z [4;34mbuild users:
2023-02-28T22:12:37.3332698Z [0m[1m    Username[0m:	UID
2023-02-28T22:12:37.3361534Z [1m     nixbld1[0m:	30001
2023-02-28T22:12:37.3372883Z [1m     nixbld2[0m:	30002
2023-02-28T22:12:37.3384524Z [1m     nixbld3[0m:	30003
2023-02-28T22:12:37.3398593Z [1m     nixbld4[0m:	30004
2023-02-28T22:12:37.3398848Z 
2023-02-28T22:12:37.3399101Z [32m[4;32mReady to continue?
2023-02-28T22:12:37.3399423Z [0mNo TTY, assuming you would say yes :)
2023-02-28T22:12:37.3399585Z 
2023-02-28T22:12:37.3399784Z [32m~~> Setting up the build group nixbld
2023-02-28T22:12:37.3591490Z [0m[1m            Created[0m:	Yes
2023-02-28T22:12:37.3616657Z 
2023-02-28T22:12:37.3617249Z [32m~~> Setting up the build user nixbld1
2023-02-28T22:12:37.3882725Z [0m[1m           Created[0m:	Yes
2023-02-28T22:12:37.3889440Z [1m            Hidden[0m:	Yes
2023-02-28T22:12:37.3910292Z [1m    Home Directory[0m:	/var/empty
2023-02-28T22:12:37.3941924Z [1m              Note[0m:	Nix build user 1
2023-02-28T22:12:37.3962458Z [1m   Logins Disabled[0m:	Yes
2023-02-28T22:12:37.3991701Z [1m  Member of nixbld[0m:	Yes
2023-02-28T22:12:37.4012359Z [1m    PrimaryGroupID[0m:	30000
2023-02-28T22:12:37.4027580Z 
2023-02-28T22:12:37.4028001Z [32m~~> Setting up the build user nixbld2
2023-02-28T22:12:37.4245054Z [0m[1m           Created[0m:	Yes
2023-02-28T22:12:37.4254275Z [1m            Hidden[0m:	Yes
2023-02-28T22:12:37.4280763Z [1m    Home Directory[0m:	/var/empty
2023-02-28T22:12:37.4311874Z [1m              Note[0m:	Nix build user 2
2023-02-28T22:12:37.4332611Z [1m   Logins Disabled[0m:	Yes
2023-02-28T22:12:37.4357839Z [1m  Member of nixbld[0m:	Yes
2023-02-28T22:12:37.4377791Z [1m    PrimaryGroupID[0m:	30000
2023-02-28T22:12:37.4392162Z 
2023-02-28T22:12:37.4392572Z [32m~~> Setting up the build user nixbld3
2023-02-28T22:12:37.4601926Z [0m[1m           Created[0m:	Yes
2023-02-28T22:12:37.4608569Z [1m            Hidden[0m:	Yes
2023-02-28T22:12:37.4630605Z [1m    Home Directory[0m:	/var/empty
2023-02-28T22:12:37.4662487Z [1m              Note[0m:	Nix build user 3
2023-02-28T22:12:37.4683373Z [1m   Logins Disabled[0m:	Yes
2023-02-28T22:12:37.4710835Z [1m  Member of nixbld[0m:	Yes
2023-02-28T22:12:37.4731927Z [1m    PrimaryGroupID[0m:	30000
2023-02-28T22:12:37.4746584Z 
2023-02-28T22:12:37.4747096Z [32m~~> Setting up the build user nixbld4
2023-02-28T22:12:37.4957502Z [0m[1m           Created[0m:	Yes
2023-02-28T22:12:37.4966728Z [1m            Hidden[0m:	Yes
2023-02-28T22:12:37.4992513Z [1m    Home Directory[0m:	/var/empty
2023-02-28T22:12:37.5026354Z [1m              Note[0m:	Nix build user 4
2023-02-28T22:12:37.5049058Z [1m   Logins Disabled[0m:	Yes
2023-02-28T22:12:37.5071807Z [1m  Member of nixbld[0m:	Yes
2023-02-28T22:12:37.5096246Z [1m    PrimaryGroupID[0m:	30000
2023-02-28T22:12:37.5096648Z 
2023-02-28T22:12:37.5097352Z [32m~~> Setting up the basic directory structure
2023-02-28T22:12:37.5185112Z [0minstall: creating directory '/nix'
2023-02-28T22:12:37.5185694Z install: creating directory '/nix/var'
2023-02-28T22:12:37.5186042Z install: creating directory '/nix/var/log'
2023-02-28T22:12:37.5186409Z install: creating directory '/nix/var/log/nix'
2023-02-28T22:12:37.5186785Z install: creating directory '/nix/var/log/nix/drvs'
2023-02-28T22:12:37.5187136Z install: creating directory '/nix/var/nix'
2023-02-28T22:12:37.5187492Z install: creating directory '/nix/var/nix/db'
2023-02-28T22:12:37.5187865Z install: creating directory '/nix/var/nix/gcroots'
2023-02-28T22:12:37.5188255Z install: creating directory '/nix/var/nix/profiles'
2023-02-28T22:12:37.5188634Z install: creating directory '/nix/var/nix/temproots'
2023-02-28T22:12:37.5189018Z install: creating directory '/nix/var/nix/userpool'
2023-02-28T22:12:37.5189428Z install: creating directory '/nix/var/nix/daemon-socket'
2023-02-28T22:12:37.5189840Z install: creating directory '/nix/var/nix/gcroots/per-user'
2023-02-28T22:12:37.5190268Z install: creating directory '/nix/var/nix/profiles/per-user'
2023-02-28T22:12:37.5266613Z install: creating directory '/nix/store'
2023-02-28T22:12:37.5340872Z install: creating directory '/etc/nix'
2023-02-28T22:12:37.5352796Z 
2023-02-28T22:12:37.5353124Z [32m~~> Installing Nix
2023-02-28T22:12:37.7121403Z [0m      Alright! We have our first nix at /nix/store/5hgddhs7c8w4qz9dxc5ndr4rm568q2lr-nix-2.14.0
2023-02-28T22:12:37.7657180Z       Just finished getting the nix database ready.
2023-02-28T22:12:37.7657862Z 
2023-02-28T22:12:37.7660465Z [32m~~> Setting up shell profiles: /etc/bashrc /etc/profile.d/nix.sh /etc/zshrc /etc/bash.bashrc /etc/zsh/zshrc
2023-02-28T22:12:37.7831301Z [0m
2023-02-28T22:12:37.7832060Z # Nix
2023-02-28T22:12:37.7832703Z if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
2023-02-28T22:12:37.7833146Z   . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
2023-02-28T22:12:37.7833401Z fi
2023-02-28T22:12:37.7833567Z # End Nix
2023-02-28T22:12:37.7833682Z 
2023-02-28T22:12:37.7995966Z 
2023-02-28T22:12:37.7996485Z # Nix
2023-02-28T22:12:37.7997297Z if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
2023-02-28T22:12:37.7997762Z   . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
2023-02-28T22:12:37.7998009Z fi
2023-02-28T22:12:37.7998188Z # End Nix
2023-02-28T22:12:37.7998304Z 
2023-02-28T22:12:37.8159052Z 
2023-02-28T22:12:37.8159507Z # Nix
2023-02-28T22:12:37.8160110Z if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
2023-02-28T22:12:37.8160546Z   . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
2023-02-28T22:12:37.8160797Z fi
2023-02-28T22:12:37.8160975Z # End Nix
2023-02-28T22:12:37.8161092Z 
2023-02-28T22:12:37.8319199Z 
2023-02-28T22:12:37.8319663Z # Nix
2023-02-28T22:12:37.8320266Z if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
2023-02-28T22:12:37.8320739Z   . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
2023-02-28T22:12:37.8320983Z fi
2023-02-28T22:12:37.8321158Z # End Nix
2023-02-28T22:12:37.8321286Z 
2023-02-28T22:12:37.8347348Z 
2023-02-28T22:12:37.8348223Z [32m~~> Setting up shell profiles for Fish with with conf.d/nix.fish inside /etc/fish /usr/local/etc/fish /opt/homebrew/etc/fish /opt/local/etc/fish
2023-02-28T22:12:37.8428144Z [0m
2023-02-28T22:12:37.8428778Z [32m~~> Setting up the default profile
2023-02-28T22:12:37.8771504Z installing 'nix-2.14.0'
2023-02-28T22:12:37.8880661Z building '/nix/store/hyq0qij16r3k6da8kp594pvl4g1nm6v5-user-environment.drv'...
2023-02-28T22:12:37.9315189Z installing 'nss-cacert-3.83'
2023-02-28T22:12:37.9425082Z building '/nix/store/k6j0a9jcvvx4pfr7dv1xss81ql5nx3hc-user-environment.drv'...
2023-02-28T22:12:37.9612233Z [0m
2023-02-28T22:12:37.9612688Z [32m~~> Setting up the nix-daemon systemd service
2023-02-28T22:12:37.9897915Z Created symlink /etc/systemd/system/nix-daemon.service → /nix/var/nix/profiles/default/lib/systemd/system/nix-daemon.service.
2023-02-28T22:12:38.2509828Z Created symlink /etc/systemd/system/sockets.target.wants/nix-daemon.socket → /nix/var/nix/profiles/default/lib/systemd/system/nix-daemon.socket.
2023-02-28T22:12:38.2512095Z Created symlink /etc/systemd/system/nix-daemon.socket → /nix/var/nix/profiles/default/lib/systemd/system/nix-daemon.socket.
2023-02-28T22:12:38.8223974Z [0m[32mAlright! We're done!
2023-02-28T22:12:38.8244531Z [0mTry it! Open a new terminal, and type:
2023-02-28T22:12:38.8244963Z 
2023-02-28T22:12:38.8245706Z   $ nix-shell -p nix-info --run "nix-info -m"
2023-02-28T22:12:38.8246035Z 
2023-02-28T22:12:38.8246452Z Thank you for using this installer. If you have any feedback or need
2023-02-28T22:12:38.8247016Z help, don't hesitate:
2023-02-28T22:12:38.8247520Z 
2023-02-28T22:12:38.8247841Z You can open an issue at
2023-02-28T22:12:38.8248463Z https://github.com/NixOS/nix/issues/new?labels=installer&template=installer.md
2023-02-28T22:12:38.8248892Z 
2023-02-28T22:12:38.8249273Z Or get in touch with the community: https://nixos.org/community
2023-02-28T22:12:38.8285331Z 
2023-02-28T22:12:38.8286072Z [34m---- Reminders -----------------------------------------------------------------
2023-02-28T22:12:38.8286733Z [0m[34m[ 1 ][0m
2023-02-28T22:12:38.8287285Z Nix won't work in active shell sessions until you restart them.
2023-02-28T22:12:38.8287666Z 
2023-02-28T22:12:41.3419777Z /home/runner/work/_actions/purcell/setup-emacs/v3.0/dist/install-nix.sh: line 53: cachix: command not found
2023-02-28T22:12:41.3522071Z ##[error]Command failed: /home/runner/work/_actions/purcell/setup-emacs/v3.0/dist/install-nix.sh emacs-28-1
2023-02-28T22:12:41.3532783Z ##[debug]Node Action run completed with exit code 1
2023-02-28T22:12:41.3536423Z ##[debug]NIX_PATH='nixpkgs=channel:nixpkgs-unstable'
2023-02-28T22:12:41.3540139Z ##[debug]Finishing: Install Emacs
```

**Comment by andschwa** [comment]
Runner info:

```
2023-02-28T22:11:39.7215408Z Requested labels: ubuntu-latest
2023-02-28T22:11:39.7215455Z Job defined at: PowerShell/PowerShellEditorServices/.github/workflows/emacs-test.yml@refs/pull/2003/merge
2023-02-28T22:11:39.7215484Z Waiting for a runner to pick up this job...
2023-02-28T22:11:40.1799542Z Job is waiting for a hosted runner to come online.
2023-02-28T22:11:42.3605490Z Job is about to start running on the hosted runner: GitHub Actions 46 (hosted)
2023-02-28T22:11:45.6973542Z ##[debug]Starting: Test via Eglot
2023-02-28T22:11:45.6999339Z ##[debug]Cleaning runner temp folder: /home/runner/work/_temp
2023-02-28T22:11:45.7334549Z ##[debug]Starting: Set up job
2023-02-28T22:11:45.7335063Z Current runner version: '2.301.1'
2023-02-28T22:11:45.7361787Z ##[group]Operating System
2023-02-28T22:11:45.7362394Z Ubuntu
2023-02-28T22:11:45.7362702Z 22.04.2
2023-02-28T22:11:45.7362940Z LTS
2023-02-28T22:11:45.7363244Z ##[endgroup]
2023-02-28T22:11:45.7363628Z ##[group]Runner Image
2023-02-28T22:11:45.7363945Z Image: ubuntu-22.04
2023-02-28T22:11:45.7364274Z Version: 20230219.1
2023-02-28T22:11:45.7364792Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20230219.1/images/linux/Ubuntu2204-Readme.md
2023-02-28T22:11:45.7365384Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230219.1
2023-02-28T22:11:45.7365891Z ##[endgroup]
2023-02-28T22:11:45.7366257Z ##[group]Runner Image Provisioner
2023-02-28T22:11:45.7366555Z 2.0.117.1
2023-02-28T22:11:45.7367046Z ##[endgroup]
```

**Comment by akirak** [comment]
It seems to be an upstream issue. `cachix/cachix-action` fails to run, so I have filed a ticket: https://github.com/cachix/cachix-action/issues/137

**Comment by pkryger** [comment]
Perhaps pinning `nix` will work as per https://github.com/cachix/cachix-action/issues/138#issuecomment-1448893176?

**Comment by purcell** [comment]
Fixed by #34, thanks.
